### DISCUSSION TITLE
Extending the Deepler plugin to translate questions is a MUST.

### DISCUSSION THREAD

**Discussion by brunobaudry** [brunobaudry/moodle-local_deepler#21] [Polls]
Currently the Deepler plugin only hepl translating course content excluding questions.

**Comment by omi1kabobi** [comment]
Hello Bruno,

I am using the DeepL plugin to translate my Moodle courses to French.
I recently opened a DeepL pro API account and input my API key to your Moodle plugin.

It was working very well for translating the course but I quickly hit the max character limit of 1 million characters only after a few dozen Moodle pages.
I increased my limit 1.33 million and hit that limit just a few minutes later, after only translating 4 pages. ( maybe 1 thousand characters)
I am translating the pages on the DeepL page one at a time and saving each to the DB as I go

I reached out to DeepL and waiting for a reply.

Do you have any idea why so many characters are being sent and charged to my DeepL Pro account?

It is starting to get expensive...

Thank you,

Omar

**Reply by brunobaudry** [reply]
Hello Omar,
Thank you for using the plugin!

I have never encountered this issue before.
I will check to see if there is a bug causing the API calls to be doubled.
As DeepL states here, character counts are based on the source code.
[Note that we are not contracted with DeepL in any way.] (https://support.deepl.com/hc/en-us/articles/360020685720-Character-count-and-billing-in-DeepL-API)
You can therefore check that the character count displayed by the plugin upon field selection matches DeepL's computed consumption after calling a translation and refreshing the page.

Example:
BEFORE SELECTION:
Total: 0 words, 0 characters (0 including spaces).
DeepL's usage: 199,612/1,000,000.
AFTER SELECTION:
Total: 4 words, 27 characters (30 characters including spaces). DeepL's usage: 199,639/1,000,000.
After calling the translation and refreshing the page:
Total: 4 words, 27 characters (30 characters including spaces). DeepL's usage: 199,642/1,000,000.

Note that the delta here is the three-space character that the plugin did not count in the estimation; this will be fixed in the next release.
However, this is not what you describe.

So please try these simple tests.
Also, check the 'Network' tab in the browser developer tools. You should see a single POST HTTP call when you press the translate button.
http://moodle.test/lib/ajax/service.php?sesskey=XXXXXX&info=local_deepler_get_translation
If you see more than one call when you click the Translate button, this could be a Moodle/plugin-related issue.

Best regards,

P.S. Please raise such comments in the appropriate GitHub section (Issues). This thread was a poll for something unrelated to your question. I will remove it soon.

**Comment by brunobaudry** [comment]
Core questions implemented. So mainstream 3rd party plugins soon.
